### PR TITLE
Improvement and small fix upon command registration code

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description

read the title

## Checklist

- [ ] - All code changes have been tested, if any
- [ ] - The documentation has been updated accordingly for any code changes
- [ ] - This PR fixes an already existing issue
- [ ] - This PR adds a new feature (new functions, methods, classes, parameters, etc.)
- [ ] - This PR fixes a bug
- [ ] - This PR is a breaking change (removes/renames functions, parameters etc.)
- [ ] - This PR is NOT a code change (changes Documentation files, README.md, CONTRIBUTING.md etc.)

### CODE REVIEW

**Review by Jerry-py — COMMENTED** [review]
This looks good but I can't approve because I don't understand anything (Rid wrote these commands and function. I'm only a mod)

File: EpikCord/__init__.py [hunk 1]
```diff
@@ -152,29 +154,16 @@ def register_slash_command(func):
...
-            )
-            self.commands[name](result)
-            return result
+    def _register_command(self, command_type, name, func):
+        command_instance = command_type(callback=func, name=name or func.__name__)
+        self.commands[name] = command_instance
+        return command_instance
 
-        return register_slash_command
+    def user_command(self, name: Optional[str] = None):
```
Comment (TheUntraceable) [inline EpikCord/__init__.py:162]: Where does this get func from?

Comment (Sigmanificient) [reply EpikCord/__init__.py:162]: from `_register_command`

Comment (TheUntraceable) [reply EpikCord/__init__.py:162]: No how do you get the callback?

Comment (TheUntraceable) [reply EpikCord/__init__.py:162]: Oh right, so the return acts like the decorators `return wrapper/inner/whatever`

**Review by TheUntraceable — COMMENTED** [review]
What about registering Slash Commands?

**Pull request closed without merging**